utils/swift_build_support/swift_build_support/products/indexstoredb.py

```diff
@@ -110,4 +105,22 @@ def run_build_script_helper(action, host_target, product, args,
     elif args.enable_tsan:
         helper_cmd.extend(['--sanitize', 'thread'])
 
+    if not product.is_darwin_host(
+            host_target) and product.is_cross_compile_target(host_target):
+        helper_cmd.extend(['--cross-compile-host', host_target])
+        build_toolchain_path = install_destdir + args.install_prefix
+        resource_dir = '%s/lib/swift' % build_toolchain_path
+        helper_cmd += [
+            '--cross-compile-config',
+            targets.StdlibDeploymentTarget.get_target_for_name(host_target).platform
+            .swiftpm_config(args, output_dir=build_toolchain_path,
+                            swift_toolchain=toolchain_path,
+                            resource_path=resource_dir)
+        ]
+
+    if action == 'install' and product.product_name() == "sourcekitlsp":
```

Comment:
Shouldn't this also be guarded by the not is_darwin_host since you are trying to not effect the Darwin path?

Reply:
While most of these changes only affect non-Darwin and are guarded by such checks, there are a few changes that are applied generally, ie for Darwin too, which I called out separately in my notes for the first commit and the second commit:

`Also, add a native_toolchain_path() method, that uses a prebuilt toolchain if available, and pass an install prefix in to swift-driver and sourcekit-lsp.`

This was needed because non-Darwin uses a different install prefix for cross-compilation hosts, so I submitted pulls for those products' build scripts [to accept a install prefix](https://github.com/apple/sourcekit-lsp/pull/424/files#diff-be797b5f5d5ebe427136b1367226887ac8db77cb2fa0709559480b456d43a460R173) or [actually start using the prefix passed in](https://github.com/apple/swift-driver/pull/743/files#diff-be797b5f5d5ebe427136b1367226887ac8db77cb2fa0709559480b456d43a460R227), which this pull now passes in for all platforms.

These configuration changes that affect Darwin too appear to be working fine.